docs/src/main/asciidoc/metrics-guide.adoc

```diff
@@ -0,0 +1,180 @@
+
+== Solution
+
+We recommend that you follow the instructions in the next sections and create the application step by step.
+However, you can go right to the completed example.
+
+Clone the Git repository: `git clone {quickstarts-clone-url}`, or download an {quickstarts-archive-url}[archive].
+
+The solution is located in the `microprofile-metrics` {quickstarts-tree-url}/microprofile-metrics[directory].
+
+== Creating the Maven Project
+
+First, we need a new project. Create a new project with the following command:
+
+[source, subs=attributes+]
+----
+mvn io.quarkus:quarkus-maven-plugin:0.11.0:create \
+    -DprojectGroupId=org.acme \
+    -DprojectArtifactId=microprofile-metrics \
+    -Dextensions="smallrye-metrics, resteasy"
+----
+
+This command generates a Maven project, importing the `smallrye-metrics` extension
+which is an implementation of the MicroProfile Metrics specification used in {project-name}.
+
+== Writing the application
+
+The application consists of a single class that implements an algorithm for checking whether a number is prime.
+This algorithm is exposed over a REST interface. Additionally, we need a few annotations to make sure
+that our desired metrics are calculated over time and can be exported for manual analysis or processing by additional tooling.
+
+The metrics that we will gather are these:
+
+* `performedChecks`: A counter which is increased by one each time the user asks about a number.
+* `highestPrimeNumberSoFar`:  This is a gauge that stores the highest number that was asked about by the user and which was determined to be prime.
+* `checksTimer`: This is a timer, therefore a compound metric that benchmarks how much time the primality tests take. We will explain in more details later.
+
+The full source code looks like this:
+
+[source,java]
+----
+package org.acme.metrics;
+
+import org.eclipse.microprofile.metrics.MetricUnits;
+import org.eclipse.microprofile.metrics.annotation.Counted;
+import org.eclipse.microprofile.metrics.annotation.Gauge;
+import org.eclipse.microprofile.metrics.annotation.Timed;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+
+@Path("/")
+public class PrimeNumberChecker {
+
+    private long highestPrimeNumberSoFar = 3;
+
+    @GET
+    @Path("/{number}")
+    @Produces("text/plain")
+    @Counted(name = "performedChecks", monotonic = true, description = "How many primality checks have been performed.")
+    @Timed(name = "checksTimer", description = "A measure how long it takes to perform the primality test.", unit = MetricUnits.MILLISECONDS)
+    public String checkIfPrime(@PathParam("number") long number) {
+        if (number < 1) {
+            return "Only natural numbers can be prime numbers.";
+        }
+        if (number == 1) {
+            return "1 is not prime.";
+        }
+        if (number % 2 == 0) {
+            return number + " is not prime, it is divisible by 2.";
```

Comment:
2 is prime.

Reply:
Good point :D I should go back to high school.